### PR TITLE
AF bugFix : removed default value 'all' from sharingFilter

### PR DESCRIPTION
**Pull request by abandonned**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/af/transform.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ function responseBuilderSimple(payload, message, destination) {
     updatedPayload.bundleIdentifier = bundleIdentifier;
   }
 
-  const sharingFilter = destination.Config.sharingFilter || "all";
+  const { sharingFilter } = destination.Config.sharingFilter;
```
Comment (saikumarrs) [inline v0/destinations/af/transform.js:92]: Did you mean to use this?
```suggestion
  const { sharingFilter } = destination.Config;
```

Comment (abandonned) [reply v0/destinations/af/transform.js]: yes

**Review by saikumarrs — CHANGES_REQUESTED** [review]
Add/update test cases for this change

**Comment by abandonned** [comment]
> Add/update test cases for this change

added and updated the old test cases

File: __tests__/data/af_input.json [hunk 2]
```diff
@@ -860,7 +876,73 @@
...
+        "AF": {
+          "af_uid": "afUid"
+        }
+      }
+    },
+    "destination": {
+      "Config": {
+        "devKey": "ef1d42390426e3f7c90ac78272e74344",
+        "androidAppId": "com.rudderlabs.javascript",
+        "sharingFilter": ["hello"]
```
Comment (saikumarrs) [inline __tests__/data/af_input.json:945]: We need to another case for default string input "all" as well.

Comment (abandonned) [reply __tests__/data/af_input.json:945]: For that case we expect it to be present in payload right?

Comment (saikumarrs) [reply __tests__/data/af_input.json:945]: No

**Review by saikumarrs — APPROVED** [review]

**Review by arnab-p — APPROVED** [review]

File: v0/destinations/af/transform.js [hunk 3]
```diff
@@ -89,7 +89,7 @@ function responseBuilderSimple(payload, message, destination) {
     updatedPayload.bundleIdentifier = bundleIdentifier;
   }
 
-  const sharingFilter = destination.Config.sharingFilter || "all";
+  const { sharingFilter } = destination.Config;
   if (isDefinedAndNotNull(sharingFilter)) {
```
Comment (akashrpo) [inline v0/destinations/af/transform.js:93]:
```suggestion
  if (isDefinedAndNotNullAndNotEmpty(sharingFilter)) {
```
We should not send sharingFilter in the payload if it is empty `''`.

**Review by saikumarrs — APPROVED** [review]

**Review by akashrpo — APPROVED** [review]

**Pull request merged**